bundles/core/src/main/java/com/adobe/cq/wcm/core/components/sandbox/internal/models/v1/TeaserImpl.java

```diff
@@ -98,39 +99,42 @@
 
     @PostConstruct
     private void initModel() {
-        withCTA = properties.get(Teaser.PN_WITH_CTA, withCTA);
-        titleValueFromPage = properties.get(Teaser.PN_TITLE_VALUE_FROM_PAGE, titleValueFromPage);
-        descriptionValueFromPage = properties.get(Teaser.PN_DESCRIPTION_VALUE_FROM_PAGE, descriptionValueFromPage);
-        linkURL = properties.get(ImageResource.PN_LINK_URL, String.class);
+        actionsEnabled = properties.get(Teaser.PN_ACTIONS_ENABLED, actionsEnabled);
+
         populateStyleProperties();
-        if (withCTA) {
-            populateCTAs();
-            if (ctas.size() > 0) {
-                ListItem firstCTA = ctas.get(0);
-                if (firstCTA.getPath() != null) {
-                    targetPage = pageManager.getPage(firstCTA.getPath());
-                }
+
+        titleFromPage = properties.get(Teaser.PN_TITLE_FROM_PAGE, titleFromPage);
+        descriptionFromPage = properties.get(Teaser.PN_DESCRIPTION_FROM_PAGE, descriptionFromPage);
+        linkURL = properties.get(ImageResource.PN_LINK_URL, String.class);
+
+        if (actionsEnabled) {
+            hiddenImageResourceProperties.add(ImageResource.PN_LINK_URL);
+            populateActions();
+            if (actions.size() > 0) {
+                ListItem firstAction = actions.get(0);
+                linkURL = firstAction.getPath();
             }
-        } else {
-            targetPage = pageManager.getPage(linkURL);
         }
-        if (hideTitle) {
+
+        targetPage = pageManager.getPage(linkURL);
+
+        if (titleHidden) {
```

Comment:
@vladbailescu, @andreeadracea - so we render the elements if they're marked hidden, but simply don't provide any text content? Shouldn't we simply not render the element at all?

Reply:
I don't think it's necessary to expose both if the title/description are hidden and their value from the model. It's enough to remove the value as rendering will take care of not showing anything, see https://github.com/Adobe-Marketing-Cloud/aem-core-wcm-components/blob/910a3060ffaa728cc98be7d1ba1c482fe5cba562/content/src/content/jcr_root/apps/core/wcm/sandbox/components/teaser/v1/teaser/title.html#L18: `data-sly-test.title="${teaser.title}"`

Reply:
I missed the `data-sly-test`, looks good